app/main.py

```diff
@@ -2,24 +1,76 @@
+from abc import ABC, abstractmethod
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+        self.value = None
+
+    def __get__(self, instance: object, owner: type) -> int:
+        return self.value
+
+    def __set__(self, instance: object, value: int) -> None:
+        if not isinstance(value, int):
+            raise TypeError("Value must be an integer.")
+        if value < self.min_amount or value > self.max_amount:
+            raise ValueError(f"Value must be between "
+                             f"{self.min_amount} and {self.max_amount}.")
+        self.value = value
+
+    def __set_name__(self, owner: type, name: str) -> None:
+        self.name = name
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    @abstractmethod
+    def validate(self, visitor: Visitor) -> bool:
+        pass
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(4, 14)
+    weight = IntegerRange(20, 50)
+    height = IntegerRange(80, 120)
+
+    def validate(self, visitor: Visitor) -> bool:
+        try:
+            self.age = visitor.age
+            self.weight = visitor.weight
+            self.height = visitor.height
+            return True
+        except (TypeError, ValueError):
+            return False
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(14, 60)
+    weight = IntegerRange(50, 120)
+    height = IntegerRange(120, 220)
+
+    def validate(self, visitor: Visitor) -> bool:
+        try:
+            self.age = visitor.age
+            self.weight = visitor.weight
+            self.height = visitor.height
```

Comment:
Similar to the `ChildrenSlideLimitationValidator`, descriptors should not be set directly. The visitor's attributes should be validated against the descriptor's constraints. Consider using the descriptor in the `Visitor` class to manage these attributes.